client/verta/verta/registry/_verta_model_base.py

```diff
@@ -93,6 +93,35 @@ def predict(self, input):
         """
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def batch_predict(self, df):
+        """Produce an output from `df`.
+
+        This method is called when batch predictions are made against a Verta endpoint.
+
+        .. note::
+
+            :meth:`batch_predict` must be written to both receive and return
+            pandas.DataFrames [1]_.
+
+            At this time, your subclass must still implement :meth:`predict`, regardless
+            of whether it contains anything meaningful.
+
+        Parameters
+        ----------
+        df : a pandas.DataFrame
+
+        Returns
+        -------
+        a pandas.DataFrame
+
+        References
+        ----------
+        .. [1] https://pandas.pydata.org/docs/reference/frame.html
+
+        """
+        pass
```

Comment:
And since it's optional, and wouldn't make sense to be called unless the user implements it, we should raise an exception if this method isn't overridden

```suggestion
        raise NotImplementedError
```

Reply:
I see! I was thinking of it the opposite way but what you're saying makes sense